src/hclust.jl

```diff
@@ -557,6 +557,11 @@ function orderbranches_barjoseph!(hmer::HclustMerges, dm::AbstractMatrix)
         (u, m) = (order[uidx], order[midx])
         (k, w) = (order[kidx], order[widx])
 
+        # Make sure node order is correct
```

Comment:
What do you think of this as an alternative? This didn't cause any errors when I ran the tests locally, so it really seems like it can't happen.

Reply:
Actually, I'd rather remove the swap from `push_merges!()` and also keep the error in the last `else` where it was before, because this check is rather artificial, and now it will happen for every node.

Reply:
Done! And confirmed that I now often hit `vr < 0` branch.